### PR TITLE
ci: Added pre-commit-hook

### LINKED ISSUE #1305: Git hook lint before commit
**Author the user story for this feature**

As i run the git add, git commit. I want lint to run to highlight issues with my code so I don't have to build the entire project or find the errors with detekt on GitHub actions.

**Describe the solution**
Add a git hook that runs either klint or detekt on git commit.

### PR DESCRIPTION
**Pull request by Sloox**
Fixes #1305

Needs to be tested locally to actually see a change. 
It requires a once off execution of the link Hook sh file.
Simply try git commit and it should run detekt.

### CODE REVIEW

**Review by piotradamczyk5 — APPROVED** [review]

**Review by jan-gogo — CHANGES_REQUESTED** [review]
The linkHooks.sh was not executable so I have changed it using `chmod +x`.
After that, I have added empty lines to random kotlin file and tested pre-commit
```
MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
hint: The '.githooks/pre-commit' hook was ignored because it's not set as executable.
hint: You can disable this warning with `git config advice.ignoredHook false`.
[#1305-git-hooks 19bd8833] Test pre-commit
 1 file changed, 9 insertions(+)
```
So I have changed `pre-commit` to executable and tried test again.
This is what I get:
```
MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
fatal: cannot run .githooks//pre-commit: No such file or directory
```
Strange

**Comment by Sloox** [comment]
> The `linkHooks.sh` was not executable so I have changed it using `chmod +x`.
> After that, I have added empty lines to a random kotlin file and tested pre-commit
> 
> ```
> MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
> hint: The '.githooks//pre-commit' hook was ignored because it's not set as executable.
> hint: You can disable this warning with `git config advice.ignoredHook false`.
> [#1305-git-hooks 19bd8833] Test pre-commit
>  1 file changed, 9 insertions(+)
> ```
> 
> So I have changed `pre-commit` to executable and tried test again.
> This is what I get:
> 
> ```
> MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
> fatal: cannot run .githooks//pre-commit: No such file or directory
> ```
> 
> Strange

Please retest it was a faulty shebang line.

**Comment by adamfilipow92** [comment]
Maybe we could add information about run ```linkHooks.sh``` before first commit to documentation in ```Environment setup``` section? It could help when for all new contributors. WDYT?

**Comment by piotradamczyk5** [comment]
One small issue on Mac
```
.githooks/linkHooks.sh: line 2: @echo: command not found
Changing git config for hooks to .githooks
Complete!
```

**Comment by piotradamczyk5** [comment]
> One small issue on Mac
> 
> ```
> .githooks/linkHooks.sh: line 2: @echo: command not found
> Changing git config for hooks to .githooks
> Complete!
> ```

I fixed it

**Review by piotradamczyk5 — APPROVED** [review]

File: .githooks/linkHooks.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+set echo of
```
Comment (piotradamczyk5) [inline .githooks/linkHooks.sh:2]:
```suggestion
set echo off
```

**Comment by piotradamczyk5** [comment]
> set echo off?

fixed

**Review by jan-gogo — APPROVED** [review]
I am not sure if `pre-commit` script will work on windows so `linkHooks.bat` probably is not necessary, but also I am not sure if any dev will use windows as the main os for development, so it's not a problem.

**Comment by Sloox** [comment]
> I am not sure if `pre-commit` script will work on windows so `linkHooks.bat` probably is not necessary, but also I am not sure if any dev will use windows as the main os for development, so it's not a problem.

I use windows and it works correctly 👍

**Pull request merged**